pkg/storage/s3/s3.go

```diff
@@ -32,22 +34,39 @@ type Storage struct {
 func New(s3Conf *config.S3Config, timeout time.Duration, options ...func(*aws.Config)) (*Storage, error) {
 	const op errors.Op = "s3.New"
 
-	creds := buildAWSCredentials(s3Conf)
-
-	awsConfig := &aws.Config{
-		Credentials: creds,
-		Region:      aws.String(s3Conf.Region),
-	}
-
+	awsConfig := aws.NewConfig()
+	awsConfig.Region = aws.String(s3Conf.Region)
 	for _, o := range options {
 		o(awsConfig)
 	}
 
-	// create a session
+	credProviders := defaults.CredProviders(awsConfig, defaults.Handlers())
+
+	if !s3Conf.UseDefaultConfiguration {
+		endpointcreds := []credentials.Provider{
+			endpointcreds.NewProviderClient(*awsConfig, defaults.Handlers(), s3Conf.CredentialsEndpoint),
+			&credentials.StaticProvider{
+				Value: credentials.Value{
+					AccessKeyID:     s3Conf.Key,
+					SecretAccessKey: s3Conf.Secret,
+					SessionToken:    s3Conf.Token,
+				},
+			},
+			&credentials.EnvProvider{},
```

Comment:
Hey, any reason to repeat EnvProvider here (given that it's already part of the default chain)?

Reply:
fixed